### ISSUE TITLE
Add Basic Streamlit UI for the project

### ISSUE THREAD

**Issue by ArunMaramraj** [Rakesh9100/ML-Project-Drug-Review-Dataset#4]
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

Hello sit. Arun here. I have worked as ML research intern for AIIMS on behalf of my college and have worked on health research domains. Thus raising this issue. This issue aims to address these concerns and proposes the following improvements to make the model more reliable and efficient.

## Handle Random Sampling:

Randomly selecting 80% of the training data may introduce bias and affect model performance. Instead, consider implementing  sampling techniques to maintain the class distribution while selecting a subset of given data.

## Optimize Memory Usage:

Use pandas' astype function to downcast numeric columns and consider using sparse data structures where applicable to further reduce memory consumption.

## HTML Decoding Function:

Evaluate the performance impact of the decode_html function that decodes HTML-encoded characters. Optimize the function, if necessary, to improve efficiency and minimize processing time.

##  Missing Value Handling:

Assess the impact of using SimpleImputer for handling missing values and explore alternative strategies such as data imputation based on domain knowledge or techniques like K-nearest neighbors (KNN) or IterativeImputer.

## Feature Encoding:

Check the effectiveness of alternative feature encoding methods such as one-hot encoding, target encoding, or entity embeddings to capture complex relationships and improve model performance.

## Regularisation
Consider implementing regularization techniques like L1 (Lasso) or L2 (Ridge) regularization in the LinearRegression and LogisticRegression models to handle potential overfitting and improve generalization.

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

**Comment by tushtithakur** [comment]
Hi, I would like to contribute. Please assign the task to me.

**Comment by ArunMaramraj** [comment]
i have created this issue to ask the mentor to assign me. You have to create your own issue

**Comment by Rakesh9100** [comment]
@ArunMaramraj Assigning you, Proceed

**Comment by ArunMaramraj** [comment]
Thank You! May i know the deadline for completion of the said issue?

**Comment by Rakesh9100** [comment]
@ArunMaramraj Update of your work pls

**Comment by SakshiRathi77** [comment]
Hello @Rakesh9100, Sakshi Rathi here.
I am ML enthusiast and exploring this domain. I would like to contribute to enhancing the speed and accuracy by adding another encoding to categorical values, normalizing, handling missing values in other ways and then providing GUI as streamlit code for the user as well as the developer.
Where the developer will be able to see charts, data visualizations, and all working functions, and at the user tab it will directly give prediction using our models with data abstraction.

**Comment by Rakesh9100** [comment]
@tushtithakur Would you like to be assigned?

**Comment by tushtithakur** [comment]
@Rakesh9100 Sure..I would like to contribute

**Comment by Rakesh9100** [comment]
Pls describe in brief what will you do in this issue @tushtithakur

**Comment by Rakesh9100** [comment]
Pls describe in brief what will you do in this issue @tushtithakur

**Comment by tushtithakur** [comment]
@Rakesh9100 Hi, I would initially split the data into 70% training and 30% test dataset. Then, I would perform data cleaning and optimize memory usage by utilizing pandas' astype function and sparse data structures. Evaluating the decode_html function's performance and exploring alternative strategies for missing value handling can improve efficiency. Additionally, exploring alternative feature encoding methods and implementing regularization techniques would enhance model performance and address overfitting. Accuracy and evaluation of the model would be conducted throughout the process.

**Comment by Rakesh9100** [comment]
Assigned you issue 11. Pls work on that @tushtithakur

**Comment by Rakesh9100** [comment]
@SakshiRathi77 Do you want to be assigned?

**Comment by SakshiRathi77** [comment]
> @SakshiRathi77 Do you want to be assigned?

Yes

**Comment by Rakesh9100** [comment]
Can you pls once brief the work you will do @SakshiRathi77

**Comment by SakshiRathi77** [comment]
AS i mentioned in above, with methods they have shared and streamlit for interface.

**Comment by Rakesh9100** [comment]
I want you to work on the Streamlit GUI part as you mentioned and provide every functionalities and visualisations for user as well as developer.
You can leave the speed and accuracy part for now and focus on GUI.
Assigning you @SakshiRathi77 Proceed

**Comment by jahnvisahni31** [comment]
I would like to work on this issue.
I have gone through the code base.✅
I understand the issue. ✅

**Comment by Rakesh9100** [comment]
Are you working? @SakshiRathi77

**Comment by Rakesh9100** [comment]
@jahnvisahni31 Do you want to work?

**Comment by ghost** [comment]
Yeah suree

**Comment by Rakesh9100** [comment]
Are you aware what you have to work upon? @Jahnvisahni
